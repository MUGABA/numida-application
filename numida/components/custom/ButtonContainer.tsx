import React from "react";
import {
  ScrollView,
  StyleProp,
  StyleSheet,
  Text,
  View,
  ViewStyle,
} from "react-native";
import PageContainer from "./PageContainer";

import BigCustomButton from "../custom/Buttons/BigCustomButton";
import Loader from "./Loader/Loader";

type Props = {
  title: string;
  children: React.ReactNode;
  buttonTitle: string;
  style?: StyleProp<ViewStyle>;
  actionFuntion: () => void;
  isLoading: boolean;
};

const ButtonContainer = ({
  title,
  children,
  buttonTitle,
  style,
  actionFuntion,
  isLoading,
}: Props) => {
  return (
    <ScrollView>
      <PageContainer>
        <View style={[styles.container, style]}>
          <Text style={styles.title}>{title}</Text>
          <View>{children}</View>
          <BigCustomButton
            buttonTitle={buttonTitle}
            actionFuntion={actionFuntion}
          />
        </View>
      </PageContainer>
      <Loader visible={isLoading} />
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 20,
  },
  title: {
    color: "black",
    fontSize: 38,
    fontWeight: "bold",
    paddingHorizontal: 12,
    paddingVertical: 25,
    marginVertical: 10,
  },
});

export default ButtonContainer;
