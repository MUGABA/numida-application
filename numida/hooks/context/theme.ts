export type Theme = {
  background: string;
  primary: string;
  secondary: string;
  text: string;
  buttonText: string;
  border: string;
};

export const lightTheme: Theme = {
  background: "#ffffff",
  primary: "#30C2E3",
  secondary: "#D4FAF5",
  text: "",
  buttonText: "",
  border: "",
};

export const darkTheme: Theme = {
  background: "#121212",
  primary: "#bb86fc",
  secondary: "#03dac6",
  text: "#ffffff",
  buttonText: "#000000",
  border: "#333333",
};
