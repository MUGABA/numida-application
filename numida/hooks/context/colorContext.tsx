// ThemeContext.tsx
import React, { createContext, ReactNode, useContext, useState } from "react";
import { darkTheme, lightTheme, Theme } from "./theme";

interface ThemeContextProps {
  theme: Theme;
  toggleTheme: () => void;
}

const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);

export const ColorThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
}) => {
  const [theme, setTheme] = useState<Theme>(lightTheme);

  const toggleTheme = () => {
    setTheme((prevTheme) =>
      prevTheme === lightTheme ? darkTheme : lightTheme
    );
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useColorTheme = (): ThemeContextProps => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error("useTheme must be used within a ThemeProvider");
  }
  return context;
};
