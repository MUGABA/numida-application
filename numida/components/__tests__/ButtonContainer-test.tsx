import { ColorThemeProvider } from "@/hooks/context/colorContext";
import { render } from "@testing-library/react-native";
import React from "react";
import { Text } from "react-native";
import ButtonContainer from "../custom/ButtonContainer";

const renderWithProvider = (ui: React.ReactElement) => {
  return render(<ColorThemeProvider>{ui}</ColorThemeProvider>);
};

describe("Button Container Component", () => {
  it(`renders correctly`, () => {
    const onClickMock = jest.fn();

    const { toJSON } = renderWithProvider(
      <ButtonContainer
        title="Dashboard"
        buttonTitle="Submit"
        actionFuntion={onClickMock}
        isLoading={false}
      >
        <Text>Another component</Text>
      </ButtonContainer>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
