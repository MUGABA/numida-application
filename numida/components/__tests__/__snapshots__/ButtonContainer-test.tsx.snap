// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Container Component renders correctly 1`] = `
<RCTScrollView>
  <View>
    <RNCSafeAreaView
      edges={
        {
          "bottom": "additive",
          "left": "additive",
          "right": "additive",
          "top": "additive",
        }
      }
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View>
          <View
            style={
              [
                {
                  "flex": 1,
                  "paddingHorizontal": 20,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontSize": 38,
                  "fontWeight": "bold",
                  "marginVertical": 10,
                  "paddingHorizontal": 12,
                  "paddingVertical": 25,
                }
              }
            >
              Dashboard
            </Text>
            <View>
              <Text>
                Another component
              </Text>
            </View>
            <View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#30C2E3",
                    "borderRadius": 15,
                    "justifyContent": "center",
                    "marginTop": 80,
                  }
                }
              >
                <Text
                  style={
                    {
                      "borderRadius": 5,
                      "color": "white",
                      "fontSize": 18,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Submit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</RCTScrollView>
`;
